cdap-security/src/main/java/co/cask/cdap/security/tools/GeneratedCertKeyStoreCreator.java

```diff
@@ -0,0 +1,169 @@
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.security.tools;
+
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.common.conf.SConfiguration;
+import com.google.common.base.Strings;
+import org.apache.commons.lang.time.DateUtils;
+import sun.security.x509.AlgorithmId;
+import sun.security.x509.CertificateAlgorithmId;
+import sun.security.x509.CertificateIssuerName;
+import sun.security.x509.CertificateSerialNumber;
+import sun.security.x509.CertificateSubjectName;
+import sun.security.x509.CertificateValidity;
+import sun.security.x509.CertificateVersion;
+import sun.security.x509.CertificateX509Key;
+import sun.security.x509.X500Name;
+import sun.security.x509.X509CertImpl;
+import sun.security.x509.X509CertInfo;
+
+import java.io.ByteArrayInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.math.BigInteger;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.security.InvalidKeyException;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.security.PrivateKey;
+import java.security.SecureRandom;
+import java.security.SignatureException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.Date;
+import java.util.Enumeration;
+
+/**
+ * Utility class with methods for generating a X.509 self signed certificate
+ * and creating a Java key store with a self signed certificate.
+ */
+public class GeneratedCertKeyStoreCreator {
+  private static final String KEY_PAIR_ALGORITHM = "RSA";
+  private static final String SECURE_RANDOM_ALGORITHM = "SHA1PRNG";
+  private static final String SECURE_RANDOM_PROVIDER = "SUN";
+
+  /* This is based on https://www.ietf.org/rfc/rfc1779.txt
+     CN      CommonName
+     L       LocalityName
+     ST      StateOrProvinceName
+     O       OrganizationName
+     OU      OrganizationalUnitName
+     C       CountryName
+     STREET  StreetAddress
+
+     All fields are not required.
+    */
+  private static final String DISTINGUISHED_NAME = "CN=App Fabric, L=Palo Alto, C=US";
+  private static final String SIGNATURE_ALGORITHM = "MD5withRSA";
+  private static final String SSL_KEYSTORE_TYPE = "JKS";
+  private static final String CERT_ALIAS = "cert";
+  private static final int KEY_SIZE = 2048;
+  private static final int VALIDITY = 999;
```

Comment:
CDAP will need to be restarted once the certificate expires. Anytime cdap master is restarted a new certificate is generated and it is valid for 999 days from that point. So, as long as there is a restart within the 999 day period we would not see an expired certificate. It is also a configurable parameter so it can be set to a longer period.